### DISCUSSION TITLE
How to paste the uploaded image?

### DISCUSSION THREAD

**Discussion by xiaogonggong-w** [toeverything/blocksuite#8754] [Q&A] [answered]
hello,
In BlockSuite, how do you paste and upload an image, not using the command to invoke the image block with a backslash?

class MockServerBlobSource implements BlobSource {
  private readonly _cache = new Map<string, Blob>();

  readonly = false;

  constructor(readonly name: string) { }

  async delete(key: string) {
    console.log(`delete ${key}`);
    
    this._cache.delete(key);
    await deleteImage(key)
  }

  async get(key: string) {
    if (this._cache.has(key)) {
      return this._cache.get(key) as Blob;
    } 
  }
  async list() {
    return Array.from(this._cache.keys());
  }

  async set(key: string, value: Blob) {
    console.log(key, value);

    this._cache.set(key, value);
    return key;
  }
}

I used this code, only implemented the `get` method, but did not use the `set` method.

**Comment by raintoway** [comment]
你还是用中文描述一遍吧，从它域复制图片进blocksuite是可以的，因为有image-adpter

**Reply by xiaogonggong-w** [reply]
好的，我现在发现的问题在于，使用 shadows: [serverblobsource] 自定义服务器的代码，在上传使用斜杠命令调取的image可以成功上传，但是直接粘贴却不行（不会走里面的set方法，而是直接调用get）。

去掉自定义上传服务器的代码，粘贴才会在笔记当中展示

**Reply by raintoway** [reply]
> 好的，我现在发现的问题在于，使用 shadows: [serverblobsource] 自定义服务器的代码，在上传使用斜杠命令调取的image可以成功上传，但是直接粘贴却不行（不会走里面的set方法，而是直接调用get）。
> 
> 去掉自定义上传服务器的代码，粘贴才会在笔记当中展示

听起来你的情况是：添加了自定义上传服务器代码后，粘贴的图片不会显示在文档中。
1. 你需要贴代码出来，要注释
2. 初步判断：你在粘贴时有报错，你可以看一下控制台，并把错误信息贴出来

**Reply by xiaogonggong-w** [reply]
> > 好的，我现在发现的问题在于，使用 shadows: [serverblobsource] 自定义服务器的代码，在上传使用斜杠命令调取的image可以成功上传，但是直接粘贴却不行（不会走里面的set方法，而是直接调用get）。
> > 去掉自定义上传服务器的代码，粘贴才会在笔记当中展示
> 
> 听起来你的情况是：添加了自定义上传服务器代码后，粘贴的图片不会显示在文档中。
> 
> 1. 你需要贴代码出来，要注释
> 2. 初步判断：你在粘贴时有报错，你可以看一下控制台，并把错误信息贴出来

https://stackblitz.com/edit/github-4vu4dz-rajahq?file=src%2Feditor%2Feditor.ts  你可以试试这个
第63行代码是关键

项目控制台报错:
![image](https://github.com/user-attachments/assets/6f35c795-2f34-4b77-aaf5-92c9229aa67a)

**Reply by raintoway** [reply, marked as answer]
> > > 好的，我现在发现的问题在于，使用 shadows: [serverblobsource] 自定义服务器的代码，在上传使用斜杠命令调取的image可以成功上传，但是直接粘贴却不行（不会走里面的set方法，而是直接调用get）。
> > > 去掉自定义上传服务器的代码，粘贴才会在笔记当中展示
> > 
> > 
> > 听起来你的情况是：添加了自定义上传服务器代码后，粘贴的图片不会显示在文档中。
> > 
> > 1. 你需要贴代码出来，要注释
> > 2. 初步判断：你在粘贴时有报错，你可以看一下控制台，并把错误信息贴出来
> 
> https://stackblitz.com/edit/github-4vu4dz-rajahq?file=src%2Feditor%2Feditor.ts 你可以试试这个 第63行代码是关键
> 
> 项目控制台报错: ![image](https://private-user-images.githubusercontent.com/79799040/387676846-6f35c795-2f34-4b77-aaf5-92c9229aa67a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwODQ3OTAsIm5iZiI6MTczMjA4NDQ5MCwicGF0aCI6Ii83OTc5OTA0MC8zODc2NzY4NDYtNmYzNWM3OTUtMmYzNC00Yjc3LWFhZjUtOTJjOTIyOWFhNjdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIwVDA2MzQ1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmZjNjZTdlZDE2YmEyNzZlYTBjYzY1YWYzZGM5NzZkNzk4M2ZkZWFmZDQ0NjMwZDA4ZGE4OGMzODU0Yjc0ZjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7MDox_SKCK0xCH3yOQIWCE0hNDq8E8zjwuDTMEouJ3Q)

```ts
const blobSources = {
    main:serverblobsource,
    // shadows: [serverblobsource],
};
```
